### DISCUSSION TITLE
Debug information for CURL HTTP Transport

### DISCUSSION THREAD

**Discussion by HLeithner** [joomla/joomla-cms#43108] [2. Cold Features]
Are developer/side builder interested in detail curl debug information?

The interface could be something like
`$transport->setDebug(true);`
or be part of the Transport creation

after request
`$transport->getDebugInformation();`

And you get the following output:

```
cUrl verbose information:
* Added example.com:443:192.0.2.1 to DNS cache
* Rebuilt URL to: https://example.com/
* Hostname was found in DNS cache
*   Trying 192.0.2.1...
* Connected to example.com (192.0.2.1) port 443 (#0)
* successfully set certificate verify locations:
*   CAfile: /lib/ssl/cacert.pem
  CApath: /etc/ssl/certs
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-SHA384
* Server certificate:
* 	 subject: C=AT; L=Wien; O=Organisation; CN=*.example.com
* 	 start date: 2024-01-27 00:00:00 GMT
* 	 expire date: 2025-01-29 23:59:59 GMT
* 	 subjectAltName: example.com matched
* 	 issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=Thawte TLS RSA CA G1
* 	 SSL certificate verify ok.
> GET / HTTP/1.1
Host: example.com
Accept: */*

< HTTP/1.1 200 OK
< activityid: 3fe1bd98-27cc-4f22-ab6c-c0c9c2bafca9
< x-forwarded-host: example.com
< x-forwarded-proto: https
< host: example.com
< x-forwarded-port: 443
< x-forwarded-for: 192.0.2.2
< accept: */*
< content-type: text/html; charset=UTF-8
< date: Thu, 21 Mar 2024 07:58:10 GMT
< transfer-encoding: chunked
< 
* Connection #0 to host example.com left intact
```

If enough people are interessiere I would bring it up in the relevant teams and create a pr for the framework

**Comment by Fedik** [comment]
Sounds as a good idea. And how about other transports?

**Reply by HLeithner** [reply]
depending on the other transports it might be possible to add similar stuff or at least basic information like resolved ip or similar but replicating the full debug functionality of curl could be a bit complex and maybe not possible for some points

**Reply by Fedik** [reply]
Yes, of course, it cannot be 1:1, they all diferent in the end.

**Reply by brianteeman** [reply]
what would be the advantage over running native curl debug tools

**Reply by Fedik** [reply]
It is about enabling "native curl debug"
